subsys/net/lib/config/init_clock_sntp.c

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2019 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_DECLARE(net_config, CONFIG_NET_CONFIG_LOG_LEVEL);
+
+#include <errno.h>
+#include <init.h>
+#include <net/sntp.h>
+#include <posix/time.h>
+
+static int init_clock_via_sntp(struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	struct sntp_time ts;
+	struct timespec tspec;
+	int res = sntp_simple(CONFIG_NET_CONFIG_SNTP_INIT_SERVER,
+			      CONFIG_NET_CONFIG_SNTP_INIT_TIMEOUT, &ts);
+
+	if (res < 0) {
+		LOG_ERR("Cannot set current time using SNTP.");
+		return res;
+	}
+
+	tspec.tv_sec = ts.seconds;
+	tspec.tv_nsec = ((u64_t)ts.fraction * (1000 * 1000 * 1000)) >> 32;
+	res = clock_settime(CLOCK_REALTIME, &tspec);
+
+	return 0;
+}
+
+SYS_INIT(init_clock_via_sntp, APPLICATION, CONFIG_NET_CONFIG_SNTP_INIT_PRIO);
```

Comment:
No need for this call. Just call `init_clock_via_sntp()` from init.c after the network is working (IP addresses set), so similar way as how network logging is initialized. Also place `init_clock_via_sntp()` directly to init.c. We also do not need separate priority as that is already handled by the init in config init.c.

Reply:
> Just call init_clock_via_sntp() from init.c after the network is working (IP addresses set), so similar way as how network logging is initialized.

Ok. I actually wanted to keep it more modular, re: discussions of possible extending that functionality to init via on-board RTC, etc., but ok, let's keep to the current functionality we have.

> Also place init_clock_via_sntp() directly to init.c.

Well, let's please keep having modular code, and not stuff everything into single gigantic file which then will be hard to work with. It's the same like existing bt_settings.c, ieee802154_settings.c - those could be put into init.c (they aren't too big either), but having them separate makes code structure much clearer.